### DISCUSSION TITLE
Re-Write  documentation

### DISCUSSION THREAD

**Discussion by marcus-j-davies** [zwave-js/node-red-contrib-zwave-js#55] [Ideas]
@crxporter,

> Meanwhile, is there anything I can do to give back to the project?

😁 

The readme for this module, is fairly big: multiple usage modes, guidance on mode usage, Managed Command Class specifications/param requirements, Managed params object structures , a sizeable change log, User Interface guide, the possible events table etc etc.

How do you feel getting involved with re-doing that (maybe branch of to Wikis - with the read me being more of an introduction)

If you prefer to get more involved on the code side - please do.

if you want to take on re-doing the documentation, I am more than happy to giving you write access to the dev branch, and adding you as a  contributor. 👍  

If so - I suggest absorbing the read me(s) as they currently are to understand what needs to be documented.

Have a think. 

Let me know if 3.4.0 works well for you - local tests were fine.

**Comment by crxporter** [comment]
Ah yes, docs. My specialty actually... play with everything then write about what it does. 

I think I can help. I won't have much time this week, but I'll be thinking about it.

**Reply by marcus-j-davies** [reply]
No worries at all, not expecting this to be an overnight thing, if you do get involved let me know, and I'll￼ add write permissions for you.

Let me know if you have any problems with 3.4.0. It should run well.

**Reply by crxporter** [reply]
I can do it with write permissions or just fork and make edits on my own branch. 

I'll be testing the new version this weekend when I get home

**Comment by marcus-j-davies** [comment]
Hey, Nothing Major.

Bit if you do device to help with re-doing documentation (again no rush what so ever), Fork/push to the dev branch. 
The dev branch is my WIP - so is usually updated with the latest documentation.

**Comment by crxporter** [comment]
So yeah it'll be really a while before I get back to this... yesterday a lightning strike took out my whole network and about half my zwave. 

So - as you could guess - I'll be spending most of my free time for the next months dealing with that.

**Reply by marcus-j-davies** [reply]
Hey man, no sweat!
Its just a "will be nice to do at some point".

But the job is yours, if you should ever want to get involved - no rush.

Oh Dam!
hope not too much damage?

A friend in the states, where he lived next door to his parents - he routed coper cables between the 2 house (networking).
His parents house, got taken out by a strike, and due to the copper cabling - it took out his entire infrastructure.

**Reply by crxporter** [reply]
Damage seems to be mostly focused on Zwave and my extensive wired network. 

Dead includes: Unifi dream machine, cams, switches, access points, one tv, all 3 Apple TV's, home theater receiver... insurance inspection is tomorrow. I hope they are generous.

**Comment by marcus-j-davies** [comment]
So,

If you're looking for some internal camera replacements, these are Apple Homekit Secure video enabled + Pan/Tilt wifi cameras at £48 GPB. - Just ordered 2 of them.

https://www.amazon.co.uk/Security-Assistant-Compatibility-Tracking-Required/dp/B086KZ8X6S

Watching/reading reviews seems that these are really really really good for the price point - given the extensive features.

They are capable of 2K Rez (1K for Secure Homekit - limitation on apples part).

Will be testing them tonight 

EDIT:
They come 2nd on iMore
https://www.imore.com/best-homekit-cameras

**Reply by crxporter** [reply]
I'll probably just replace them for exactly what I had before - unifi cams. I like having the onsite storage and it's easy enough to get them in homekit using nodered! 

I think the only change I want to make compared to what got destroyed is a different home theater receiver. Any favorites for 7.1 in the theater and stereo in a second zone? Bonus points if zone 2 works with airplay 2!

**Comment by crxporter** [comment]
Update from Texas: big stack of Zwave 700 toys just showed up! It'll be party time tomorrow.

**Reply by marcus-j-davies** [reply]
Hey!

You had me at `big stack of Zwave 700 toys` 😆 .
See, I'm still rolling with the 500 series of devices (I should really invest in updating them all).

So, how are things - insurance paid up?.

Well, we have seen a few version updates since your outage (mainly Z-Wave JS pushing out some revisions).
so when the time comes to re-installing everything, see below for the latest (which brings you up to 3.7.1)

![image](https://user-images.githubusercontent.com/55892693/120064277-45c89280-c063-11eb-96c5-1c896a0c210f.png)


https://github.com/zwave-js/node-red-contrib-zwave-js/blob/main/CHANGELOG.md

**Reply by crxporter** [reply]
Insurance is... trying I guess. The guy who came out to check my things had no idea how to handle such fancy toys. Everything is settled except for the Unifi switches. He thought it would be fine to match a 24 port managed PoE switch from Unifi with the cheapest available 24 port switch from netgear. 

But with zwave 700, when I was shopping I noticed the 700 switches were on sale for $22 each which is actually cheaper than any of the 500 options so I figured it was time to begin my upgrade! And I threw in a 700 usb stick, too. 

Problem with the "upgrade everything" idea is all my things aren't yet available on 700 chips. So it'll be a slow burn over the next year or more. 

Now I just need to muster enough energy to go work on the switch boxes.

**Reply by marcus-j-davies** [reply]
I have 2 Aeotec Range Extenders, which seem to run fine my modest setup ~ 30 nodes
I think my idea of wanting to go 700, is more the excitement of going 700, as opposed to gaining any benefit

**Reply by crxporter** [reply]
Is it faster? What should I be watching for?

Honestly I just got it cause it was $22 per switch instead of like $30 each for the ones I had before.

**Reply by marcus-j-davies** [reply]
There are 3 benefits.
 - Lower power usage
 - Greater Range
 - Faster Die

You will see benefits from the range, and if you have an intricate network setup, you will likely see some benefits in speed (especially when all devices are routing off of each other) - but that is when all devices are 700.

If devices are mixed between 500/700 - you will see a better range, but maybe not so much with speed.

**Reply by marcus-j-davies** [reply]
You will see benefits regardless I think - due to the range, as that will have an impact on speed as well.
the more hops a node takes to reach the controller, the slower things are.

so if a device can reach the controller without hopping, naturally will be faster.

and of course - the golden rule,
once your nodes are in place, and paired to the controller, do a network heal (waking up those that are battery operated)

**Reply by crxporter** [reply]
The 700 setup is alive and well. Only had to make 2 very small changes for my code to work with the new switches, which apparently are sending `duration` updates to the same command class as on/off or dimmer level... 

No real noticeable changes so far. I'm at 50 devices including the controller so I don't know that I ever hit traffic levels where more speed is needed. Sure does have me itching for more 700 devices though 🤦‍♂️

**Reply by marcus-j-davies** [reply]
Hurray!

I bet its nice to have it all back up?

are you running the latest plugin version (3.7.1)
There has been a few fixes to the underlining Z-Wave driver - not sure if what was fixed, the 50 possible nodes were affected anyway - but you never know.

**Reply by marcus-j-davies** [reply]
>which apparently are sending duration updates to the same command class as on/off

The **BinarySwitch** & **MultiLevelSwitch** zwave classes do support the `Duration` object.
But some devices, prefer having a set duration in a config param or something to that affect, instead of including it in the Set command its self, which means they may only support older versions of the classes

```javascript
let MyDuration = {
  Duration: {
    value: 5,
    unit: "seconds" | "minutes",
  }
}

let Message = {
    payload: {
        node: 17,
        class: "MultiLevelSwitch",
        operation: "Set",
        endpoint: 1, // only needed for some multi switch devices (defaults to 0)
        params: [75, MyDuration]
    }
}
return Message
```

**Reply by crxporter** [reply]
Yes it's great to be back running. 

I did update to 3.7.1 before the changeover, everything went smoothly. I just have to learn to use the new devices a bit. 

Side note- this is the first time I've done 100% of the setup inside of nodered, my last stick was actually programmed originally using openzwave then ran in zwavejs2mqtt and finally nodered. I enjoyed doing it all fresh from the nodered interface. 

And finally for duration, yes I send duration like that. For some reason though when controlled at the switch, my new dimmers are sending a duration message back to nodered. Not a problem, I just put a condition to ignore that message in my flows.

**Reply by marcus-j-davies** [reply]
>I enjoyed doing it all fresh from the nodered interface.

nice to hear - though I cant take credit for that one, I am however, looking to tidy up the UI at some point, to.

 - Cash in on some enhancements with the node i have made
 - Introduce Association mapping.
